ios/FileUtils.m

```diff
@@ -77,29 +81,80 @@ @implementation FileUtils
                   rejecter:(RCTPromiseRejectBlock)reject
                   )
 {
-    UIImage *image = [UIImage imageWithContentsOfFile:path];
-    NSData* imageData =  UIImageJPEGRepresentation(image, 1.0);
-    CGImageSourceRef sourceRef = CGImageSourceCreateWithData((CFDataRef)imageData, NULL);
-    
-    NSDictionary *metadata = (__bridge NSDictionary *)CGImageSourceCopyPropertiesAtIndex(sourceRef,0,NULL);
-    NSDictionary *exif = [metadata objectForKey:(NSString *)kCGImagePropertyTIFFDictionary];
-    NSDictionary *datetime = [exif objectForKey:(NSString *)kCGImagePropertyExifDateTimeOriginal];
-    
-    resolve(datetime);
-}
+    // Path for getting exif from asset id if image or creation date for video
+    if(![path hasPrefix:@"file:///"]) {
+        PHAsset* asset = [PHAsset fetchAssetsWithLocalIdentifiers:@[path] options:nil].firstObject;
+        PHContentEditingInputRequestOptions *editOptions = [[PHContentEditingInputRequestOptions alloc]init];
+        editOptions.networkAccessAllowed = YES;
+        
+        // If image, use exif data
+        if ([type isEqualToString:@"image"]) {
+            [asset requestContentEditingInputWithOptions:editOptions completionHandler:^(PHContentEditingInput *contentEditingInput, NSDictionary *info) {
+                CIImage *image = [CIImage imageWithContentsOfURL:contentEditingInput.fullSizeImageURL];
+                NSDictionary *properties = image.properties;
+                NSDictionary *exif = [properties objectForKey:(NSString *)kCGImagePropertyExifDictionary];
+                NSDictionary *datetime = [exif objectForKey:(NSString *)kCGImagePropertyExifDateTimeOriginal];
+                resolve(datetime);
+                return;
+            }];
+        
+        // If not an image, get last modified date
+        } else {
+            resolve(asset.creationDate);
+            return;
+        }
 
-@end
+    // Path for getting exif from file path if image or creation date for video
+    } else {
+        NSString *prefixToRemove = @"file:///";
+        NSString *pathWithoutFilePrefix = [path copy];
+        if ([path hasPrefix:prefixToRemove])
+            pathWithoutFilePrefix = [path substringFromIndex:[prefixToRemove length]];
+        
+        // If image, use exif data
+        if ([type isEqualToString:@"image"]) {
+            NSData* fileData = [NSData dataWithContentsOfFile:pathWithoutFilePrefix];
+            CGImageSourceRef mySourceRef = CGImageSourceCreateWithData((CFDataRef)fileData, NULL);
+            if (mySourceRef != NULL)
+            {
+                NSDictionary *properties = (__bridge NSDictionary *)CGImageSourceCopyPropertiesAtIndex(mySourceRef,0,NULL);
+                NSDictionary *exif = [properties objectForKey:(NSString *)kCGImagePropertyExifDictionary];
+                NSDictionary *datetime = [exif objectForKey:(NSString *)kCGImagePropertyExifDateTimeOriginal];
+                resolve(datetime);
+                return;
+            }
```

Comment:
Some duplication here, we may want to extract this out into a private helper method